assembly/index.ts

```diff
@@ -1,4 +1,221 @@
-import { JSONDecoder } from './decoder';
+import { JSONDecoder, JSONHandler } from './decoder';
 import { JSONEncoder } from './encoder';
 
+export namespace JSON {
+    export enum Val_Type {
```

Comment:
s/Val_Type/JSONType/

Reply:
I mentioned this below, but since it's namespaced that would make this `json.JSONType`.  I choose `Val` since the JSON spec defines these as JSON values.